### PR TITLE
Main Page UI for Search function

### PR DESCRIPTION
**Pull request by mishrapalak**
Page has search button and area for table but other buttons still need to be added

### CODE REVIEW

**Review by miralsumrani — DISMISSED** [review]

**Comment by mishrapalak** [comment]
Yes those 3 need to be added. Currently working on a file to connect heroku to this html file.

**Review by miralsumrani — COMMENTED** [review]
Consider using "get" method to pull checkfridge query from index.js rather than hardcoding similar query in

**Comment by nandatanvi** [comment]
Looks great. Will be adding "Find out what is expiring soon button", the ability to see today's date, and other details to this main page in my pull request.

File: index.js [hunk 1]
```diff
@@ -17,7 +17,12 @@ express()
   .use(express.static(path.join(__dirname, 'public')))
   .set('views', path.join(__dirname, 'views'))
   .set('view engine', 'ejs')
-  .get('/', (req, res) => res.render('pages/index'))
+  .get('/', jsonParser, async (req, res) => {
+    try {
+      const client = await pool.connect();
+      const result = await client.query(`SELECT * FROM fridge_products`);
+      const results = { 'results': (result) ? result.rows : null};
+      res.render('pages/index', results );
```
Comment (mdschnei) [inline index.js:25]: It looks like you might be missing a few lines here. The try/catch statements are opened here but aren't closed, which will likely lead to syntax errors as the code is expecting the try statement to be closed with a curly bracket and a subsequent catch statement to follow. Perhaps this accidentally got cut off somewhere when copy pasting the template from a different endpoint?

Regardless, try adding the following code just after line 26 to close the try/catch statements and resolve the syntax error:
```
client.release();
    } catch (err) {
      console.error(err);
      res.send("Error " + err);
    }
})
```

File: views/pages/db.ejs [hunk 2]
```diff
@@ -16,6 +16,10 @@
       <ul>
           <% results.forEach(function(r) { %>
               <li><%= r.id %> - <%= r.name %></li>
+              <td><%= r.prod_name%></td>
+              <td><%= r.prod_type%></td>
+              <td><%= r.prod_expiry%></td>
+              <td><%= r.prod_quantity%></td>
```
Comment (mdschnei) [inline views/pages/db.ejs]: The changes in this file aren't actually necessary to commit as it's just a testing file, and also the "prod_xxx" parameters won't actually exist for this results object since it refers to the test_table table which just has "id" and "name" as keys. Could you delete these four lines between 19 and 22, and then recommit this file to reverse the changes?

File: views/pages/index.ejs [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
+  <tr>
+    <th>Products</th>
+    <th>Type</th>
+    <th>Date of Expiry</th>
+    <th>Quantity</th>
+  </tr>
+            <td><%= r.prod_name%></td>
+            <td><%= r.prod_type%></td>
+            <td><%= r.prod_expiry%></td>
+            <td><%= r.prod_quantity%></td>
```
Comment (mdschnei) [inline views/pages/index.ejs]: This is great, but it looks like there might be just a couple missing elements here. Firstly, the "results" variable passed to this page by the server needs to be accessed in a forEach loop first before you can access its fields using r.xxx in the table. For reference you can take a look at the db.ejs file which has a very similar operation. Adding the line `<% results.forEach(function(r) { %>` before the <td> elements will allow you to access the fields of results through the variable r, as you are doing here. After that don't forget to close the forEach loop using the closing tag `<% }); %>` below the <td> elements.

Additionally, I believe there should be a <tr> tag before the <td> elements (but still inside the forEach loop) as well as a corresponding </tr> closing tag after them (also still inside the forEach loop). That way, each time the loop runs to create a new item entry it will be creating the cells on a new row.

And lastly, it looks like according to the database structure for the fridge_products and products tables that we are querying from, the key names are actually 'prod_name', 'type', 'exp_dt', and 'qty', in order. Updating these should ensure that the correct object values are being read and printed into the table.

So to summarize, the final version should look something like this:

```
<% results.forEach(function(r) { %>
  <tr>
    <td><%= r.prod_name%></td>
    <td><%= r.type%></td>
    <td><%= r.exp_dt%></td>
    <td><%= r.qty%></td>
  </tr>
<% }); %>
```

File: index.js [hunk 4]
```diff
@@ -17,7 +17,12 @@ express()
   .use(express.static(path.join(__dirname, 'public')))
   .set('views', path.join(__dirname, 'views'))
   .set('view engine', 'ejs')
-  .get('/', (req, res) => res.render('pages/index'))
+  .get('/', jsonParser, async (req, res) => {
+    try {
+      const client = await pool.connect();
+      const result = await client.query(`SELECT * FROM fridge_products`);
```
Comment (mdschnei) [inline index.js]: Here's an updated query that you can use here instead of `SELECT * FROM fridge_products` that will let you also access the product name and type (since those only exist in the product table):
```
const result = await client.query(`SELECT prod_name, type, exp_dt, qty FROM fridge_products NATURAL JOIN products`);
```

You can just replace line 23 with that above ^^

File: index.js (hunk 1 above)
Comment (mishrapalak) [reply index.js:25]: good catch, thank you!

File: views/pages/db.ejs (hunk 2 above)
Comment (mishrapalak) [reply views/pages/db.ejs]: forgot to remove these, thank you!

File: views/pages/index.ejs (hunk 3 above)
Comment (mishrapalak) [reply views/pages/index.ejs]: Thanks for the clarification, I have added this.

File: index.js (hunk 4 above)
Comment (mishrapalak) [reply index.js]: Thank you!
